coreMacros/src/main/scala/Chisel/internal/sourceinfo/SourceInfoTransform.scala

```diff
@@ -111,10 +111,9 @@ class SourceInfoTransform(val c: Context) extends SourceInfoTransformMacro {
     */
   def doFuncTerm = {
     val funcName = c.macroApplication match {
-      case Apply(TypeApply(Select(_, TermName(funcName)), _), _) => funcName
-      case Apply(Select(_, TermName(funcName)), _) => funcName
-      case Select(_, TermName(funcName)) => funcName
-
+      case q"$_.$funcName(..$_)" => funcName
+      case q"$_.$funcName[$_](..$_)" => funcName
+      case q"$_.$funcName" => funcName
```

Comment:
This much more clearly illustrates what is being checked.
Also, I think it should be: (note the ... over ..)
case q"$_.$funcName[..$_](...$_)" => funcName
case q"$_.$funcName(...$_)" => funcName

Please add:
case _ => throw new Exception(s"Chisel Internal Error: Could not resolve function name from macro application: ${c.showCode(c.macroApplication)}")
Remember, this will be thrown by scalac during scalac runtime (code compilation)

Actually, you should be able to get away with just:
```scala
val funcName = c.macroApplication match {
  case q"$_.$funcName[..$_](...$_)" => funcName
  case _ => throw new Exception(s"Chisel Internal Error: Could not resolve function name from macro application: ${c.showCode(c.macroApplication)}")
```
You may need q"$_.$funcName", unsure. Need to test.